Lib/test/test_httpservers.py

```diff
@@ -1536,6 +1541,261 @@ def test_server_test_ipv4(self, _):
+                self.invoke_httpd(tls_password_option, self.tls_password_file)
+                mock_func.reset_mock()
+
+        # Abnormal: --tls-password-file cannot be opened
+
+        non_existent_file = 'non_existent_file'
+        for tls_password_option in tls_password_options:
+            for tls_cert_option in tls_cert_options:
+                with self.assertRaises(SystemExit):
+                    self.invoke_httpd(tls_cert_option,
+                                      self.tls_cert,
+                                      tls_password_option,
+                                      non_existent_file)
+
+    @mock.patch('http.server.test')
+    def test_no_arguments(self, mock_func):
+        self.invoke_httpd()
+        mock_func.assert_called_once_with(HandlerClass=self.default_handler,
+                                          ServerClass=self.default_server,
+                                          protocol=self.default_protocol,
+                                          port=self.default_port,
+                                          bind=self.default_bind,
+                                          tls_cert=None,
+                                          tls_key=None,
+                                          tls_password=None)
+        mock_func.reset_mock()
+
+    @mock.patch('http.server.test')
+    def test_help_flag(self, _):
+        options = ['-h', '--help']
+        for option in options:
+            with self.assertRaises(SystemExit):
+                output = self.invoke_httpd(option)
+                self.assertStartsWith(output, 'usage: ')
+
+    @mock.patch('http.server.test')
+    def test_unknown_flag(self, _):
+        with self.assertRaises(SystemExit):
+            output = self.invoke_httpd('--unknown-flag')
+            self.assertStartsWith(output, 'usage: ')
+
+
+class CommandLineRunTimeTestCase(unittest.TestCase):
+    random_data = os.urandom(1024)
+    random_file_name = 'random.bin'
+    tls_cert = certdata_file('ssl_cert.pem')
+    tls_key = certdata_file('ssl_key.pem')
+    tls_password = 'somepass'
+
+    def setUp(self):
+        super().setUp()
+        with open(self.random_file_name, 'wb') as f:
+            f.write(self.random_data)
+        self.addCleanup(os_helper.unlink, self.random_file_name)
+        self.tls_password_file = tempfile.mktemp()
+        with open(self.tls_password_file, 'wb') as f:
+            f.write(self.tls_password.encode())
+        self.addCleanup(os_helper.unlink, self.tls_password_file)
+
+    def fetch_file(self, path, allow_self_signed_cert=True) -> bytes:
+        context = ssl.create_default_context()
+        if allow_self_signed_cert:
+            context.check_hostname = False
+            context.verify_mode = ssl.CERT_NONE
+        req = urllib.request.Request(path, method='GET')
+        res = urllib.request.urlopen(req, context=context)
+        return res.read()
+
+    def parse_cli_output(self, output: str) -> tuple[str, str, int]:
```

Comment:
```suggestion
    def parse_cli_output(self, output):
```

Please check that ALL type annotations are removed.